osu.Game/Overlays/Chat/ChatLine.cs

```diff
@@ -240,6 +240,9 @@ public MenuItem[] ContextMenuItems
             {
                 get
                 {
+                    if (sender.Id == User.SYSTEM_USER.Id)
```

Comment:
Any reason for comparing the `Id` rather than the user itself?

Reply:
No particular reason other than making it similar to this line:
https://github.com/ppy/osu/blob/73590bfca1a4131581aa74c6d7b28a8378729221/osu.Game/Overlays/Chat/ChatLine.cs#L251

I just checked that `User` is `IEquatable<User>` so it might also be better to change both perhaps?

Reply:
Yep, I think that's fine.

Reply:
Direct comparison doesn't work for user-sent messages as the first image was sent in a previous session and the later was sent in the current session. Likely due to reference inequality.
![image](https://user-images.githubusercontent.com/20495991/123191425-cf894580-d4d3-11eb-8581-6d18a809a9a3.png)
![image](https://user-images.githubusercontent.com/20495991/123191472-e2037f00-d4d3-11eb-93e3-d6c2122d6b8a.png)

Just to be sane I'll only change the check against the system user.

Reply:
The equality comparer checks non-reference equality. Are you sure you're using `.Equals()`?